src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -5911,8 +5981,14 @@ static
 MonoInst*
 arch_emit_simd_intrinsics (const char *class_ns, const char *class_name, MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsig, MonoInst **args)
 {
+	if (!strcmp (class_ns, "System.Runtime.Intrinsics.X86")) {
```

Comment:
Is there a similar line for `System.Runtime.Intrinsics.Arm` and `System.Runtime.Intrinsics.Wasm` missing?

Reply:
Handled elsewhere in that source file.

Reply:
You will find Arm and Wasm under respective defines.

Reply:
Do they not need a path to ensure that `IsSupported` returns constant `false` and the intrinsics directly generate a `PNSE` exception, rather than hitting the recursive fallback, or is that handled elsewhere as well?

Reply:
I mainly focused on x86 in this PR, other code will still go through the fallbacks, but could probably be enhanced as well.